geoengine/auth.py

```diff
@@ -35,14 +36,26 @@ class Session:
 
     session: ClassVar[req.Session] = None
 
-    def __init__(self, server_url: str) -> None:
+    def __init__(self, server_url: str, credentials: Tuple[str, str] = None) -> None:
         '''
         Initialize communication between this library and a Geo Engine instance
+
+        if credentials are provided, the session will be authenticated
+
+        optional arguments: (email, password) as tuple
+        optional environment variables: GEOENGINE_EMAIL, GEOENGINE_PASSWORD
         '''
 
-        # TODO: username and password for Pro version
+        session = None
 
-        session = req.post(f'{server_url}/anonymous').json()
+        if credentials is not None:
+            session = req.post(f'{server_url}/login', json={"email": credentials[0], "password": credentials[1]}).json()
+        elif "GEOENGINE_EMAIL" in os.environ and "GEOENGINE_PASSWORD" in os.environ:
```

Comment:
i guess the order should be (credentials in code, file, env)

Reply:
c2f62fda29c12bdb8d5ce304e5c95e5f6e3cec58